pkg/cloudprovider/drift.go

```diff
@@ -0,0 +1,75 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/providers/amifamily"
+	"github.com/aws/karpenter/pkg/utils"
+)
+
+func (c *CloudProvider) isNodeFromDriftedNodeTemplate(ctx context.Context, machine *v1alpha5.Machine, provisioner *v1alpha5.Provisioner, nodeTemplate *v1alpha1.AWSNodeTemplate) (bool, error) {
```

Comment:
nit: This naming doesn't make a _ton_ of sense to me. Why not just, `isNodeTemplateDrifted`? I assume you are doing all of this breaking up so that you can do an `isDrifted` call on each aspect of the spec?

Reply:
I was thinking that would be confusing because that seems like we are checking if an AWSNodeTemplate is drifted, as opposed to checking if a machine is drifted against a NodeTemplate. Would `isMachineFromDriftedNodeTemplate` be better?

Reply:
And yes all the calls  from the NodeTemplate spec will be placed in that function